src/main/java/org/opensearch/ubi/action/UserBehaviorInsightsActionFilter.java

```diff
@@ -72,42 +70,78 @@ public void onResponse(Response response) {
                 // Get the search itself.
                 final SearchRequest searchRequest = (SearchRequest) request;
 
-                // TODO: Restrict logging to only queries of certain indices specified in the settings.
-                //final List<String> indices = Arrays.asList(searchRequest.indices());
-                //final Set<String> indicesToLog = new HashSet<>(Arrays.asList(settings.get(SettingsConstants.INDEX_NAMES).split(",")));
-                //if(indicesToLog.containsAll(indices)) {
-
                 // Get all search hits from the response.
                 if (response instanceof SearchResponse) {
 
                     // Get info from the headers.
                     final String queryId = getHeaderValue(HeaderConstants.QUERY_ID_HEADER, UUID.randomUUID().toString(), task);
-                    final String eventStore = getHeaderValue(HeaderConstants.EVENT_STORE_HEADER, "default", task);
+                    final String eventStore = getHeaderValue(HeaderConstants.EVENT_STORE_HEADER, "", task);
                     final String userId = getHeaderValue(HeaderConstants.USER_ID_HEADER, "", task);
                     final String sessionId = getHeaderValue(HeaderConstants.SESSION_ID_HEADER, "", task);
 
-                    // The query will be empty when there is no query, e.g. /_search
-                    final String query = searchRequest.source().toString();
+                    // If there is no event store header, ignore this search.
+                    if(!"".equals(eventStore)) {
+
+                        // Get the id_field to use for each result's unique identifier.
+                        final String queriesIndexName = UbiUtils.getQueriesIndexName(eventStore);
+                        final GetSettingsRequest getSettingsRequest = new GetSettingsRequest().indices(queriesIndexName);
+
+                        final GetSettingsResponse getSettingsResponse = client.admin().indices().getSettings(getSettingsRequest).actionGet();
```

Comment:
Changing this to a read-through cache of the index settings should be done at some point and captured in #92.